f5_openstack_agent/lbaasv2/drivers/bigip/lbaas_driver.py

```diff
@@ -15,158 +15,161 @@
 
 
 class LBaaSBaseDriver(object):
-    """Abstract base LBaaS Driver class for interfacing with Agent Manager """
+    """Abstract base LBaaS Driver class for interfacing with Agent Manager."""
 
     def __init__(self, conf):  # XXX 'conf' appears to be unused
-        '''Maybe we can remove this method altogether? Or maybe it's for future
+        """Maybe we can remove this method altogether? Or maybe it's for future.
 
         subclassing...
-        '''
+        """
         self.agent_id = None
         self.plugin_rpc = None  # XXX overridden in the only known subclass
         self.connected = False  # XXX overridden in the only known subclass
         self.service_queue = []
         self.agent_configurations = {}  # XXX overridden in subclass
 
     def set_context(self, context):
-        """Set the global context object for the lbaas driver """
+        """Set the global context object for the lbaas driver."""
         raise NotImplementedError()
 
-    def post_init(self):
-        """Run after agent is fully connected """
-        raise NotImplementedError()
+    def set_plugin_rpc(self, plugin_rpc):
+        """Provide LBaaS Plugin RPC access."""
 
-    def set_tunnel_rpc(self, tunnel_rpc):  # XXX into this class?
-        """Provide FDB Connector RPC access """
+    def set_agent_report_state(self, report_state_callback):
+        """Set Agent Report State."""
         raise NotImplementedError()
 
-    def set_l2pop_rpc(self, l2pop_rpc):
-        """Provide FDB Connector with L2 Population RPC access """
+    def set_tunnel_rpc(self, tunnel_rpc):
+        """Provide FDB Connector RPC access."""
         raise NotImplementedError()
 
-    def connect(self):
-        """Connect backend API endpoints """
+    def set_l2pop_rpc(self, l2pop_rpc):
+        """Provide FDB Connector with L2 Population RPC access."""
         raise NotImplementedError()
 
     def flush_cache(self):
-        """Remove all cached items """
+        """Remove all cached items."""
         raise NotImplementedError()
 
+    def backend_integrity(self):
+        """Return True, if the agent is be considered viable for services."""
+        raise NotImplemented()
+
     def backup_configuration(self):
-        """Persist backend configuratoins """
+        """Persist backend configuratoins."""
         raise NotImplementedError()
 
-    def get_stats(self, service):
-        """Get Stats for a loadbalancer Service """
-        raise NotImplementedError()
+    def generate_capacity_score(self, capacity_policy):
+        """Generate the capacity score of connected devices."""
+        raise NotImplemented
```

Comment:
Shouldn't we raise NotImplementedError() and not NotImplemented? NotImplemented is a Python constant (AFAIK), and the exception class name is NotImplementedError. Someone correct me if I'm wrong. I know this is from the previous code and not introduced in these changes, but we might want to correct this.

Reply:
This was on there before.  I don't recall changing this.